aspnetcore/mvc/controllers/routing.md

```diff
@@ -70,667 +65,788 @@ public class HomeController : Controller
+>
+> * `id` is set to `0` by model binding.
+> * No entity is found in the database matching `id == 0`.
+>
+> [Attribute routing](#ar) provides fine-grained control to make the ID required for some actions and not for others. By convention, the documentation includes optional parameters like `id` when they're likely to appear in correct usage.
 
-Routing will only match a combination of action and controller that are defined by the app. This is intended to simplify cases where conventional routes overlap.
+Enable [Logging](xref:fundamentals/logging/index) to see how the built-in routing implementations, such as <xref:Microsoft.AspNetCore.Routing.Route>, match requests.
 
-## Multiple routes
+[Attribute routing](#ar) is explained later in this document.
 
-You can add multiple routes inside `UseEndpoints` by adding more calls to `MapControlleRoute`. Doing so allows you to define multiple conventions, or to add conventional routes that are dedicated to a specific action, such as:
+<a name="mr"></a>
 
-```csharp
-app.UseEndpoints(endpoints =>
-{
-   endpoints.MapControllerRoute("blog", "blog/{*article}",
-            defaults: new { controller = "Blog", action = "Article" });
-   endpoints.MapControllerRoute("default", "{controller=Home}/{action=Index}/{id?}");
-});
-```
+### Multiple conventional routes
 
-The `blog` route here is a *dedicated conventional route*, meaning that it uses the conventional routing system, but is dedicated to a specific action. Since `controller` and `action` don't appear in the route template as parameters, they can only have the default values, and thus this route will always map to the action `BlogController.Article`.
+Multiple [conventional routes](#cr) can be added inside `UseEndpoints` by adding more calls to <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllerRoute*> and <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapAreaControllerRoute*>. Doing so allows defining multiple conventions, or to adding conventional routes that are dedicated to a specific [action](#action), such as:
+
+[!code-csharp[](routing/samples/3.x/main/Startup.cs?name=snippet_1)]
 
-Adding routes using `MapControllerRoute` defines a priority order for the routing system to respect. Matches from a route that appears earlier will have a higher priority. So in this example, the `blog` route will have a higher priority for matches than the `default` route.
+<a name="dcr"></a>
 
-> [!NOTE]
-> *Dedicated conventional routes* often use catch-all route parameters like `{*article}` to capture the remaining portion of the URL path. This can make a route 'too greedy' meaning that it matches URLs that you intended to be matched by other routes. Put the 'greedy' routes later in the route table to solve this.
+The `blog` route in the preceding code is a **dedicated conventional route**. It's called a dedicated conventional route because:
+
+* It uses the [conventional routing](#cr) system.
```

Comment:
We should not describe conventional routing (nor attribute routing) as a **system**. There is one routing system, and those are the ways of configuring it. 

I'd suggest dropping the word `system`. 

(this is a mistake I should have caught in my rewrite, legacy routing kinda had two systems)